### PR TITLE
fixed_address support for iobar

### PR DESCRIPTION
**Pull request by GHswitt**
mmiobar has fixed_address support. Add it for iobar.

### CODE REVIEW

File: chipsec/hal/iobar.py [hunk 1]
```diff
@@ -88,6 +88,10 @@ def get_IO_BAR_base_address( self, bar_name ):
             # this method is not preferred
             base = self.cs.pci.read_word( int(bar['bus'],16), int(bar['dev'],16), int(bar['fun'],16), int(bar['reg'],16) )
 
+        if 'fixed_address' in bar:
+            base = int(bar['fixed_address'],16)
+            if logger().VERBOSE: logger().log('[iobar] Using fixed address for {}: 0x{:016X}'.format(bar_name, base))
+
```
Comment (ErikBjorge) [inline chipsec/hal/iobar.py]: You should only override the base address if you were not able to read it from a register or PCI configuration space. I would expect a check for 0xFFFF as well as checking for the fixed address attribute.

Comment (GHswitt) [reply chipsec/hal/iobar.py]: Ok, how about the latest commit?

File: chipsec/hal/iobar.py [hunk 2]
```diff
@@ -82,11 +82,18 @@ def get_IO_BAR_base_address( self, bar_name ):
             if 'base_field' in bar:
                 base_field = bar['base_field']
                 base = self.cs.read_register_field( bar_reg, base_field, preserve_field_position=True )
+                empty_base = self.cs.get_register_field( bar_reg, 0xFFFFFFFF, base_field, preserve_field_position=True)
```
[the remark this answers is not in the document]
Comment (GHswitt) [reply chipsec/hal/iobar.py]: Ok

File: chipsec/hal/iobar.py [hunk 3]
```diff
@@ -82,11 +82,18 @@ def get_IO_BAR_base_address( self, bar_name ):
...
             else:
                 base = self.cs.read_register( bar_reg )
+                empty_base = 0xFFFF
```
[the remark this answers is not in the document]
Comment (GHswitt) [reply chipsec/hal/iobar.py]: Ok

**Pull request merged**